TicTacToe/GamePlay.cs

```diff
@@ -0,0 +1,100 @@
+using System;
+
+namespace TicTacToe
+{
+    public class GamePlay
+    {
+        private int _player;
+        private GameStatus _gameStatus;
+        private readonly Board _board;
+        private readonly int[,] _boardArray;
+        private bool _givenUp;
+        
+        public GamePlay()
+        {
+            _board = new Board();
+            _boardArray = _board.GetBoard();
+            _player = RandomlyPickPlayerForFirst();
+            _givenUp = false;
+            _gameStatus = GameStatus.Playing;
+        }
+
+        public void Play()
+        {
+            var uiConsole = new UserInputConsole();
+            while (_gameStatus is GameStatus.Playing)
+            {
+                uiConsole.OutputBoard(_boardArray);
+                uiConsole.UpdatePlayerInput(_player);
+                if (uiConsole.PlayerHasGivenUp())
+                {
+                    _givenUp = true;
+                }
+                else
+                {
+                    var inputMove = uiConsole.GetPlayerMove();
+                    string outputMessage = HandleMoveInput(inputMove);
+                    uiConsole.OutputMessage(outputMessage);
+                }
+                UpdateGameStatus();
```

Comment:
I think UpdateGameStatus() implies there is a change every time the loop is run, when actually it stays as Playing until, the player quits, there is a win or a draw. So unless a player quits you are going through 5 loops at a minimum before making a change. you also have this _givenUp guy that could be used to exit the loop